cmake/SelectHTTPSBackend.cmake

```diff
@@ -54,7 +54,7 @@ if(USE_HTTPS)
 
 		set(GIT_OPENSSL 1)
 		list(APPEND LIBGIT2_SYSTEM_INCLUDES ${OPENSSL_INCLUDE_DIR})
-		list(APPEND LIBGIT2_SYSTEM_LIBS ${OPENSSL_LIBRARIES})
+		list(APPEND LIBGIT2_SYSTEM_LIBS ${OPENSSL_LIBRARIES} ${CMAKE_DL_LIBS})
```

Comment:
Since this doesn't seem to be an issue when dynamically linked, maybe this should be guarded by a check for `LINK_WITH_STATIC_LIBRARIES`?